libraries/freertos_plus/standard/freertos_plus_tcp.cmake

```diff
@@ -0,0 +1,40 @@
+afr_module(NAME "freertos_plus_tcp" INTERNAL)
+
+set(src_dir "${CMAKE_CURRENT_LIST_DIR}/freertos_plus_tcp")
+set(inc_dir "${CMAKE_CURRENT_LIST_DIR}/freertos_plus_tcp/include")
+set(test_dir "${CMAKE_CURRENT_LIST_DIR}")
+
+# Create a list of all FReeRTOS+TCP headers files in the FreeRTOS+TCP
+# library.
+# The list of header files will be added to metadata required
+# for the FreeRTOS console.
+file(GLOB_RECURSE INCLUDE_HEADER_FILES "${inc_dir}/*.h")
+
+afr_module_sources(
+    ${AFR_CURRENT_MODULE}
+    PRIVATE
+        "${src_dir}/FreeRTOS_ARP.c"
+        "${src_dir}/FreeRTOS_DHCP.c"
+        "${src_dir}/FreeRTOS_DNS.c"
+        "${src_dir}/FreeRTOS_IP.c"
+        "${src_dir}/FreeRTOS_Sockets.c"
+        "${src_dir}/FreeRTOS_Stream_Buffer.c"
+        "${src_dir}/FreeRTOS_TCP_IP.c"
+        "${src_dir}/FreeRTOS_TCP_WIN.c"
+        "${src_dir}/FreeRTOS_UDP_IP.c"
+        # List of files added to the target so that these are available
+        # in code downloaded from the FreeRTOS console.
+        ${CMAKE_CURRENT_LIST_DIR}/freertos_plus_tcp.cmake
+        ${INCLUDE_HEADER_FILES}
+)
+
+afr_module_include_dirs(
+    ${AFR_CURRENT_MODULE}
+    PUBLIC "${inc_dir}"
+    PRIVATE "${test_dir}"
```

Comment:
Why does FreeRTOS+TCP module depend on test?

Reply:
@AniruddhaKanhere, do you know the reason for it? 
I received build failure in previous commit without the include path dependency to `tests/integration_test`

```
..\..\..\libraries\freertos_plus\standard\freertos_plus_tcp\FreeRTOS_DNS.c(1901): fatal error C1083: Cannot open include file: 'freertos_tcp_test_access_dns_define.h': No such file or directory
```

Reply:
It should be in the include path for `afr_test_module(freertos_tcp_test)`.

Reply:
Please ignore the above comment. I realize that the reason is - FreeRTOS_DNS.c includes freertos_tcp_test_access_dns_define.h when the FreeRTOS+TCP is built for test. The file freertos_tcp_test_access_dns_define.h exists in the tests directory and therefore it needs to be in the include path.

Reply:
To make it clear that test dir inclusion is only needed when building for test, I'd suggest the following:

```
afr_module_include_dirs(
    ${AFR_CURRENT_MODULE}
    PUBLIC "${inc_dir}"
)
if(AFR_ENABLE_TESTS)
    afr_module_include_dirs(
        ${AFR_CURRENT_MODULE}
        PRIVATE
            "${AFR_TESTS_DIR}/integration_test"
    )
endif()
```